hvpy/tests/test_io.py

```diff
@@ -9,3 +10,24 @@ def test_default_get_output_type_is_raw():
 def test_url_property():
     params = HvpyParameters()
     assert params.url == "https://api.helioviewer.org/v2//"
+
+
+def test_set_url():
+    # This is not safe if we ever decide to parallelize the tests.
+    params_first = HvpyParameters()
+    assert params_first.url == "https://api.helioviewer.org/v2//"
+
+    set_api_url("https://api.beta.helioviewer.org/")
+    params_second = HvpyParameters()
+    assert params_second.url == "https://api.beta.helioviewer.org//"
+
+    assert params_first.url == params_second.url
+    set_api_url("https://api.helioviewer.org/v2/")
+
+
+def test_set_url_env(env):
+    env.set("HELIOVIEWER_API_URL", "https://fake_env_url/")
+
+    from hvpy.config import Settings
+
+    assert Settings().api_url == "https://fake_env_url/"
```

Comment:
Can we add one more assertion after this that if you call `set_api_url` it uses the url passed in instead of the environment?

Reply:
I can do that but I am not sure it would be useful here. 

The `set_api_url` function will not change this specific url since this `Setting` is not affected by that by the LiveSettings.